### ISSUE TITLE
Global installation doesn't work

### ISSUE THREAD

**Issue by ygoe** [babel/babel#7381]
This is a bug report.

Installing babel globally, it either does nothing or fails with an error.

### Input Code

Just a basic sample.

```js
var array = [0, 1, 2];
array.filter(v => v);
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

Not using this.

### Expected Behavior

I'd like old-style JavaScript code. Especially nothing with `=>` in it.

### Current Behavior

The code remains unchanged if I just call `babel new.js`.

When calling `babel new.js --presets=env`, and after installing babel-preset-env (also globally of course), I see this error message:

> Error: Couldn't find preset "env" relative to directory "."

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  6.26.0
| node             | 8.9.4
| npm              | 5.6.0
| Operating System | Windows 10

**Comment by takahiro-saeki** [comment]
@ygoe Hi! I was wondering why you wanna use it as a global dependency. I tried to be occured your error, then I reproduced your error.
unfortunately, I couldn't resolve this error though, I simply suggest you to use as a local dependency.

### devDependencies
- @babel/cli": "^7.0.0-beta.40"
- @babel/core": "^7.0.0-beta.40"
- @babel/preset-env": "^7.0.0-beta.40"

### sample javascript file.
sample.js
```js
const sample = data => param => data + param;
```

### just do it in terminal
`$ npx babel sample.js -d lib`

that's it. you can generate sample.js which is transformed by babel to be inside of lib directory.

**Comment by loganfsmyth** [comment]
Installing Babel globally isn't really something we support. You can certainly install the CLI there, but plugins and presets are always resolved relative to the working directory / config file / file being compiled, which means plugins and presets installed globally won't be found.

**Comment by ygoe** [comment]
Why I want to install it globally? Because there is no "local" here. I don't have JavaScript projects as you probably imagine them. I don't even know what that is or how they look like. I'm using babel as a tool. And it can be used for any files I want. It's supposed to _work_, not to _work for a single project_. I'm working with PHP website projects and ASP.NET Core web application projects developed with Visual Studio. You probably never do that. In these projects I need it for the front-end stuff.

Actually I have found a workaround now. IIRC I need to specify the full path of the preset directory with is somewhere in %AppData%\npm\node_modules, not just its name. That makes the command line very long, but it doesn't matter because it's all wrapped in a GUI tool anyway.

I need something that converts newer JavaScript code to older code for IE, as long as it's used and supported by its maker. babel seems to do that well. So I've chosen it as backend in my web compiler tool, together with others in the pipeline such as rollup, uglifyjs (sorry, babel-minify broke my code), sassc and csso.

**Comment by lemmon** [comment]
Yeah, I get why local installation is preferred but for when you want to transform a few scripts for smaller projects here and there it's not quite effective to install 225 packages and 4419 files for every one of them.

Global support would be a nice to have feature.

**Comment by lemmon** [comment]
I have hacked together a small CLI utility that installs babel core and preset-env globally. Might not be for everybody, it's built for my needs to compile small scripts fast without the need for local installation.

https://github.com/lemmon/babel-env-cli

**Comment by s-m-e** [comment]
Just ran into this issue ...

> Installing Babel globally isn't really something we support. [...]

I am modernizing a tool-chain that used to integrate `babel-cli` and `babel-preset-env`, both apparently now superseded by new package structures. I am therefore now instead using (or attempting to use) `@babel/cli`, `@babel/core` and `@babel/preset-env`. Just like I did it before, I am trying to install those packages "globally", i.e. via `npm install -g X`. **However**, I am using `npm` and `node` from within a [Python virtual environment](https://docs.python.org/3/library/venv.html), integrated via [nodeenv](https://github.com/ekalinin/nodeenv). So "global" really is not actually really global in my case but rather "global within the [currently activated] virtual environment".

> Why I want to install it globally? Because there is no "local" here. I don't have JavaScript projects as you probably imagine them. I don't even know what that is or how they look like. I'm using babel as a tool. And it can be used for any files I want. It's supposed to work, not to work for a single project. 

This pretty much summarizes my use-case as well. Ironically, I did work before. I am using this Python virtual environment and `babel` within it to maintain a number of statically generated websites. Having `babel` and its plugins installed in a "global" context allows me to have all of `babel`'s components relevant to me in one place instead of numerous copies of this software scattered across all project folders. In a sense, `babel` for me is one CLI tool among many ... I'd love if this use-case could be supported. 

I'd be more than happy to test patches if there were any. I'd also help if I could but my understanding of this ecosystem is too limited I suppose.

@lemmon Great tool, but for whatever reason not working for my use-case. Your `babel-env` throws the exact same error as the original `babel` command. Happy to file a report if this is within the scope of your project.

**Comment by s-m-e** [comment]
I should add that there is a **workaround** for my use-case, in case this helps someone: Linking the `node-modules` into my various project directories. Not ideal, but still better than keeping a ton of copies of `babel` and plugins around.

```bash
ln -s /path/to/python/env/lib64/node_modules node_modules
```

**Comment by liuxingbaoyu** [comment]
> I should add that there is a **workaround** for my use-case, in case this helps someone: Linking the `node-modules` into my various project directories. Not ideal, but still better than keeping a ton of copies of `babel` and plugins around.

Have you seen this? I think it might help.
https://yarnpkg.com/features/pnp

**Comment by ygoe** [comment]
> I am modernizing a tool-chain (...)

Funny, I did that, too, recently – and threw out babel completely. Nobody needs IE-compatible code anymore, all browsers now support modern JavaScript. Finally I got rid of that burden.
